port/linuxaarch64/omrsignal_context.c

```diff
@@ -0,0 +1,318 @@
+
+#define NGPRS 32
+
+/* AArch64 Signal Context Reserved Space Size */
+#define RESERVED_SPACE_SZ 4096
+
+void
+fillInUnixSignalInfo(struct OMRPortLibrary *portLibrary, void *contextInfo, struct OMRUnixSignalInfo *j9Info)
+{
+	j9Info->platformSignalInfo.context = (ucontext_t *)contextInfo;
+	/* module info is filled on demand */
+}
+
+uint32_t
+infoForSignal(struct OMRPortLibrary *portLibrary, struct OMRUnixSignalInfo *info, int32_t index, const char **name, void **value)
+{
+	*name = "";
+
+	switch (index) {
+
+	case OMRPORT_SIG_SIGNAL_TYPE:
+	case 0:
+		*name = "J9Generic_Signal_Number";
+		*value = &info->portLibrarySignalType;
+		return OMRPORT_SIG_VALUE_32;
+
+	case OMRPORT_SIG_SIGNAL_PLATFORM_SIGNAL_TYPE:
+	case 1:
+		*name = "Signal_Number";
+		*value = &info->sigInfo->si_signo;
+		return OMRPORT_SIG_VALUE_32;
+
+	case OMRPORT_SIG_SIGNAL_ERROR_VALUE:
+	case 2:
+		*name = "Error_Value";
+		*value = &info->sigInfo->si_errno;
+		return OMRPORT_SIG_VALUE_32;
+
+	case OMRPORT_SIG_SIGNAL_CODE:
+	case 3:
+		*name = "Signal_Code";
+		*value = &info->sigInfo->si_code;
+		return OMRPORT_SIG_VALUE_32;
+
+	case OMRPORT_SIG_SIGNAL_HANDLER:
+	case 4:
+		*name = "Handler1";
+		*value = &info->handlerAddress;
+		return OMRPORT_SIG_VALUE_ADDRESS;
+
+	case 5:
+		*name = "Handler2";
+		*value = &info->handlerAddress2;
+		return OMRPORT_SIG_VALUE_ADDRESS;
+	
+	case OMRPORT_SIG_SIGNAL_INACCESSIBLE_ADDRESS:
+	case 6:
+		/* si_code > 0 indicates that the signal was generated by the kernel */
+		if (info->sigInfo->si_code > 0) {
+			if ((info->sigInfo->si_signo == SIGBUS) || (info->sigInfo->si_signo == SIGSEGV)) {
+				*name = "InaccessibleAddress";
+				*value = &info->sigInfo->si_addr;
+				return OMRPORT_SIG_VALUE_ADDRESS;
+			}
+		}
+		return OMRPORT_SIG_VALUE_UNDEFINED;
+
+	// case OMRPORT_SIG_SIGNAL_ADDRESS:
+	// case 7:
+	// 	*name = "fault_address";
+	// 	*value = &((struct sigcontext *)&info->platformSignalInfo.context->uc_mcontext)->fault_address;
+	// 	return OMRPORT_SIG_VALUE_ADDRESS;
+
+
+	default:
+		return OMRPORT_SIG_VALUE_UNDEFINED;
+	}
+}
+
+
+/**
+ * Walk the 4k reserve space to get the floating point registers since the layout differ per-chip
+ */
+void*
+walkReserveSpace(uint32_t magic_to_find, uint8_t *reserved_space)
+{
+	if(!reserved_space) {
+		return NULL;
+	}
+	
+	int i=0;
+	while(i<RESERVED_SPACE_SZ) {
+		struct _aarch64_ctx *header = (struct _aarch64_ctx *)&reserved_space[i];
+		if(!header || 0 < header->size) {
+			return NULL;
+		} else if(header->magic == magic_to_find) {
+			return (void*)&reserved_space[i];
+		}
+		i += header->size;
+	}
+	return NULL;
+}
+
+uint32_t
+infoForFPR(struct OMRPortLibrary *portLibrary, struct OMRUnixSignalInfo *info, int32_t index, const char **name, void **value)
+{
+	*name = "";
+	
+	/**
+	 * per documentation these are 128 bits registers, can be used as 64, 32 and 16 bit floats
+	 * (16 bit floats CANNOT be used for arithmetic)
+	 */
+	 
+	if ((index >= 0) && (index < NGPRS)) {
```

Comment:
Shouldn't you define a `NFPRS` for strict correctness?  Yes, it works out that NFPRS == NGPRS but it makes it very clear there is a distinction.

Reply:
I was sticking with the PPC template since the registers are alike and I wanted to stay consistent. But I agree it should be split!